src/components/Search/Search.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import { px } from '../../utils';
+import classnames from 'classnames';
+import Input from '../Input/Input';
+import SearchIcon from '../../assets/search.svg?react';
+
+export interface SearchProps extends React.HTMLAttributes<HTMLElement> {
+  altText?: string;
+  useIcon?: boolean;
+}
+
+const Search = ({ altText = 'Search', useIcon = true }: React.PropsWithChildren<SearchProps>) => {
+  const [overlayEnabled, setOverlayEnabled] = React.useState(false);
+
+  const showSearch = () => {
+    setOverlayEnabled(!overlayEnabled);
+  };
+
+  return (
+    <div className={`${px}-search`}>
+      <div
+        data-testid="search-overlay"
+        className={classnames(`${px}-search__overlay`, { [`${px}-search__overlay--active`]: overlayEnabled })}
+        onClick={showSearch}
+      />
+      <button data-testid="search-button" className={`${px}-search__button`} onClick={showSearch}>
```

Comment:
I think once we convert this to Remix we'll need to make this a `<Form` so probably the `<div` at the top will need to support an ability to pass a custom element.